src/components/molecules/Pagination/index.vue

```diff
@@ -0,0 +1,143 @@
+<template>
+  <ul class="pagination-inner">
+    <template
+      v-if="disabledLinkToFirst"
+    >
+      <li>
+        <app-router-link
+          :to="'?page=1'"
+          large
+          white
+          bg-theme-color
+          hover-opacity
+        >
+          1
+        </app-router-link>
+      </li>
+      <li
+        class="pagination-item"
+      >
+        <app-text
+          ellipsis
+          bold
+        >
+          …
+        </app-text>
+      </li>
+    </template>
+    <li
+      v-for="page in pageRange"
+      :key="page"
+      class="pagination-item"
+    >
+      <template
+        v-if="page === currentPage"
+      >
+        <app-button
+          big
+          disabled
+        >
+          {{ page }}
+        </app-button>
+      </template>
+      <template
+        v-else
+      >
+        <app-router-link
+          :to="`?page=${page}`"
+          large
+          white
+          bg-theme-color
+          hover-opacity
+        >
+          {{ page }}
+        </app-router-link>
+      </template>
+    </li>
+    <template
+      v-if="disabledLinkToLast"
+    >
+      <li
+        class="pagination-item"
+      >
+        <app-text
+          ellipsis
+          bold
+        >
+          …
+        </app-text>
+      </li>
+      <li
+        class="pagination-item"
+      >
+        <app-router-link
+          :to="`?page=${totalPages}`"
+          large
+          white
+          bg-theme-color
+          hover-opacity
+        >
+          {{ totalPages }}
+        </app-router-link>
+      </li>
+    </template>
+  </ul>
+</template>
+
+<script>
+import { Text, RouterLink, Button } from '@Components/atoms';
+
+export default {
+  components: {
+    appText: Text,
+    appRouterLink: RouterLink,
+    appButton: Button,
+  },
+  props: {
+    currentPage: {
+      type: Number,
+      default: null,
+    },
+    totalPages: {
+      type: Number,
+      default: null,
+    },
+  },
+  computed: {
+    disabledLinkToFirst() {
+      return this.currentPage > 3;
+    },
+    disabledLinkToLast() {
+      return this.currentPage < this.totalPages - 2;
+    },
+    pageRange() {
+      const { currentPage, totalPages } = this;
+      const pages = [];
+      for (let i = 1; i < totalPages + 1; i += 1) {
+        pages.push(i);
+      }
```

Comment:
[imo]
配列はfor文を使わなくても作成することができます！

```suggestion
      const totalPageNumberArray = [...Array(this. totalPages)].map((_, i) => i + 1);
```

Reply:
ありがとうございます！この配列作成方法は初めて知りました。
次回以降生かしていきます！